app/controllers/countries_controller.rb

```diff
@@ -3,4 +3,16 @@ def show
     @country = Country.find(params[:id])
     @cities = City.where(country_id: params[:id]).order("name")
   end
+
+  def index
+    if params[:q]
+      if params[:q] == ''
+        redirect_to '/', notice: "You must input a character"
+      else
+        @city = params[:q]
+        redirect_to :controller => 'search', :action => 'search', :q => @city
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.